Refactor FeatureComplete code for color support and formatting help text

rodrigoprimo opened the pull request:
This PR moves the code related to checking color and formatting the help text to a new helper class. This way, it can be reused when the new DocCodeExamples script is introduced.

Tests were added for the new HelpTextFormatter::format() method.

This commit was originally added to #185, but after discussing it with @jrfnl, I'm opening a separate PR for it.

jrfnl commented on Scripts/Utils/HelpTextFormatter.php:
```diff
@@ -0,0 +1,137 @@
+     *
+     * @return bool
+     */
+    public static function isColorSupported()
```
I don't think this method belongs in the `HelpTextFormatter` class as this is also used to determine whether the non-help _output_ of the CLI command(s) should be colorized or not...

I have a feeling this should be in a separate class, which maybe should also contain some "colorize" helper methods.

rodrigoprimo replied:
As we discussed in a call, I will update this PR to remove `isColorSupported()` and then look into potentially creating a separate PR to add https://github.com/php-parallel-lint/PHP-Console-Color/ as a dependency and use it in this package.

jrfnl commented on Scripts/Utils/HelpTextFormatter.php:
```diff
@@ -0,0 +1,102 @@
+    /**
+     * Format help text from a structured array of options.
+     *
+     * @param array<string, array<array<string, string>>> $helpTexts   The help texts to format.
```
As the formatter is now in a separate class, I think the expected format for the `$helpTexts` array needs to be documented in more detail.

The class docblock might even be a better place than this function docblock.

rodrigoprimo replied:
I added documentation of the `$helpTexts` array format to the class docblock as suggested.

jrfnl reviewed (commented):
Thanks for setting this up @rodrigoprimo !

A couple of questions about the "design":
1. Is the `PHPCSDevTools\Scripts\FeatureComplete\Config::getHelp()` method still necessary ?
2. Any particular reason why the `HelpTextFormatter` is a static class and not an instantiatable one ?
3. As the logic for the help display has now moved to the `HelpTextFormatter`, I wonder how much value the tests in the `Tests\FeatureComplete\GetHelpTest` now have and whether the `@covers` for that test is still correct/useful.

rodrigoprimo commented:
Thanks for the feedback, @jrfnl!

> A couple of questions about the "design":

I addressed all three points:
1. Removed `getHelp()` and inlined the `HelpTextFormatter` call directly in `processCliCommand()`.
2. Converted `HelpTextFormatter` to an instantiable class.
3. Removed `GetHelpTest` entirely.

Besides that, before implementing the changes you suggested, I rebased this branch with the latest from `develop`, force pushed and then fixed a few CS violations due to new rules introduced after the last time I worked on this PR.
